flash/data/batch.py

```diff
@@ -0,0 +1,186 @@
+from typing import Any, Callable, Mapping, Optional, Sequence, Union
+
+import torch
+from pytorch_lightning.trainer.states import RunningStage
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+from flash.data.utils import _contains_any_tensor, convert_to_modules
+
+
+class _Sequential(torch.nn.Module):
+    """
+    This class is used to chain 3 functions together for the _Preprocessor ``per_sample_transform`` function.
+    1. ``per_sample_pre_tensor_transform``
+    2. ``per_sample_to_tensor_transform``
+    3. ``per_sample_post_tensor_transform``
+    """
+
+    def __init__(
+        self,
+        per_sample_pre_tensor_transform: Callable,
+        per_sample_to_tensor_transform: Callable,
+        per_sample_post_tensor_transform: Callable,
+        assert_contains_tensor: bool = False
+    ):
+        super().__init__()
+
+        self.per_sample_pre_tensor_transform = convert_to_modules(per_sample_pre_tensor_transform)
+        self.per_sample_to_tensor_transform = convert_to_modules(per_sample_to_tensor_transform)
+        self.per_sample_post_tensor_transform = convert_to_modules(per_sample_post_tensor_transform)
+        self.assert_contains_tensor = assert_contains_tensor
+
+    def forward(self, sample: Any):
+        sample = self.per_sample_pre_tensor_transform(sample)
+        sample = self.per_sample_to_tensor_transform(sample)
+        if self.assert_contains_tensor:
+            if not _contains_any_tensor(sample):
+                raise MisconfigurationException(
+                    "When ``per_sample_to_tensor_transform`` is overriden, "
+                    "``DataPipeline`` expects the outputs to be ``tensors``"
+                )
+        sample = self.per_sample_post_tensor_transform(sample)
+        return sample
+
+    def __str__(self) -> str:
+        repr_str = f'{self.__class__.__name__}:'
+        repr_str += f'\n\t\t(per_sample_pre_tensor_transform): {repr(self.per_sample_pre_tensor_transform)}'
+        repr_str += f'\n\t\t(per_sample_to_tensor_transform): {repr(self.per_sample_to_tensor_transform)}'
+        repr_str += f'\n\t\t(per_sample_post_tensor_transform): {repr(self.per_sample_post_tensor_transform)}'
+        repr_str += f'\n\t\t(assert_contains_tensor): {repr(self.assert_contains_tensor)}'
+        return repr_str
+
+
+class _PreProcessor(torch.nn.Module):
+    """
+        This class is used to encapsultate the following functions of a Preprocess Object:
+        Inside a worker:
+            per_sample_transform: Function to transform an individual sample
+                Inside a worker, it is actually make of 3 functions:
+                    * per_sample_pre_tensor_transform
+                    * per_sample_to_tensor_transform
+                    * per_sample_post_tensor_transform
+            collate: Function to merge sample into a batch
+            per_batch_transform: Function to transform an individual batch
+                * per_batch_transform
```

Comment:
```suggestion

                * per_batch_transform
```
is this a single element list?